repos/config/detail/lock_repos.py

```diff
@@ -71,7 +117,10 @@ def fetch_repo_details(url, branch):
             raise ValueError(result.stderr)
         commit_hash, time = result.stdout.split("/")
         os.chdir(cwd)
-    return commit_hash, time
+    return {
+        "hash":commit_hash,
+        "shallow_since": time,
+    }
```

Comment:
@kilian-funk same thing here, this could be a class probably to make it easier to extend later on if needed, and be explicit about what this function returns.

Reply:
Will leave it for now.